apps/web/app/api/code/submission/route.ts

```diff
@@ -0,0 +1,124 @@
+import { NextRequest, NextResponse } from "next/server";
+import db from "@repo/db/client";
+
+const getInputString = (args: string[], funcName: string) => {
+  return `\n
+console.log(${funcName}(${args.join(",")}))
+`;
+};
+
+let successfulSubmissions: any[] = [];
+
+async function fetchSubmissions(tokenString: string, resolve: (value: any[] | PromiseLike<any[]>) => void) {
+  // Fetch submissions with the tokenString
+  const subRes = await fetch(`${process.env.NEXT_PUBLIC_JUDGE0_API_URL}/submissions/batch?tokens=${tokenString}`);
+
+  // Parse the response
+  const newData = await subRes.json();
+
+  // Separate submissions based on status id
+  const processingSubmissions = newData.submissions.filter((submission: any) => submission.status.id <= 2);
+  const processedSubmissions = newData.submissions.filter((submission: any) => submission.status.id > 2);
+
+  // Move submissions with status id > 2 to another array
+  successfulSubmissions = [...successfulSubmissions, ...processedSubmissions];
+
+  // Check if any moved submissions with status id > 2
+  if (processingSubmissions.length > 0) {
+    // Create a new token string for submissions with status id <= 2
+    tokenString = processingSubmissions.map((submission: any) => submission.token).join(",");
+
+    // Fetch new submissions recursively after a delay
+    setTimeout(() => fetchSubmissions(tokenString, resolve), 1000); // Fetch data after 5 seconds
+  } else {
+    return resolve(processedSubmissions);
+  }
+}
+
+export async function POST(req: NextRequest) {
+  const { sourceCode, languageId, problemStatementId } = await req.json();
+
+  const problemStatement = await db.problemStatement.findFirst({
+    where: { id: problemStatementId },
+    include: {
+      testCases: true,
+    },
+  });
+
+  if (!problemStatement) {
+    return NextResponse.json({ message: "Problem Statement not found" }, { status: 400 });
+  }
+
+  const { testCases, mainFuncName } = problemStatement;
+
+  const submissions = testCases.map((testCase) => {
+    return {
+      source_code: atob(sourceCode) + getInputString(testCase.inputs, mainFuncName),
```

Comment:
Do we need to do atob on the `getInoutString` as well?

Reply:
Storing the code as base64_encoded, as in the admin will store it as btoa(), so converting it back to atob to append the input string.